app/services/foreman_vault/vault_auth_method.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module ForemanVault
+  class VaultAuthMethod
+    ALLOWED_COMMON_NAMES = ['<FQDN>'].freeze
+
+    def initialize(host)
+      @host = host
+    end
+
+    def valid?
+      name.present? && options[:certificate].present?
+    end
+
+    def name
+      return if !host || !vault_policy_name
+
+      [host, vault_policy_name].join('-').parameterize
+    end
+
+    def save
+      return false unless valid?
+
+      set_certificate(name, options)
+    end
+
+    def delete
+      return false unless name
+
+      delete_certificate(name)
+    end
+
+    private
+
+    attr_reader :host
+    delegate :vault_policy, :vault_connection, to: :host
+    delegate :name, to: :vault_policy, prefix: true
+    delegate :set_certificate, :delete_certificate, to: :vault_connection
+
+    def options
+      {
+        certificate: certificate,
+        token_policies: name,
```

Comment:
I guess we need to use `vault_policy_name` here as that's the name the of the policy.
```suggestion
        token_policies: vault_policy_name,
```